src/classes/ffhttp_Client.cls

```diff
@@ -29,86 +29,310 @@
+
+	// Similar to 401 Unauthorized, but it indicates that the client needs to authenticate itself in 
+	// order to use a proxy.
+	public static final Integer HTTP_STATUS_CODE_ERROR_PROXY_AUTHENTICATION_REQUIRED = 407;
+
+	// The server did not receive a complete request message within the time that it was prepared 
+	// to wait.
+	public static final Integer HTTP_STATUS_CODE_ERROR_REQUEST_TIMEOUT = 408;
+
+	// The request could not be completed due to a conflict with the current state of the target 
+	// resource. This code is used in situations where the user might be able to resolve the conflict 
+	// and resubmit the request.
+	public static final Integer HTTP_STATUS_CODE_ERROR_CONFLICT = 409;
+
+	// The target resource is no longer available at the origin server and that this condition is 
+	// likely to be permanent.
+	public static final Integer HTTP_STATUS_CODE_ERROR_GONE = 410;
+
+	// The server refuses to accept the request without a defined Content-Length.
+	public static final Integer HTTP_STATUS_CODE_ERROR_LENGTH_REQUIRED = 411;
+
+	// One or more conditions given in the request header fields evaluated to false when tested 
+	// on the server.
+	public static final Integer HTTP_STATUS_CODE_ERROR_PRECONDITION_FAILED = 412;
+
+	// The server is refusing to process a request because the request payload is larger than the 
+	// server is willing or able to process.
+	public static final Integer HTTP_STATUS_CODE_ERROR_REQUEST_ENTITY_TOO_LARGE = 413;
+
+	// The server is refusing to service the request because the request-target is longer than the 
+	// server is willing to interpret.
+	public static final Integer HTTP_STATUS_CODE_ERROR_REQUEST_URI_TOO_LONG = 414;
+
+	// The origin server is refusing to service the request because the payload is in a format not
+	// supported by this method on the target resource.
+	public static final Integer HTTP_STATUS_CODE_ERROR_UNSUPPORTED_MEDIA_TYPE = 415;
+
+	// None of the ranges in the request's Range header field1 overlap the current extent of the 
+	// selected resource or that the set of ranges requested has been rejected due to invalid ranges 
+	// or an excessive request of small or overlapping ranges.
+	public static final Integer HTTP_STATUS_CODE_ERROR_REQUESTED_RANGE_NOT_SATISFIABLE = 416;
+
+	// The expectation given in the request's Expect header field could not be met by at least one 
+	// of the inbound servers.
+	public static final Integer HTTP_STATUS_CODE_ERROR_EXPECTATION_FAILED = 417;
+
+	// Curl command
+	private static final String CURL_COMMAND = 'curl';
+	private static final String CURL_REQUEST_METHOD = ' -X ';
+	private static final String CURL_HEADER = ' -H ';
+	private static final String CURL_BODY = ' -d ';
+	private static final String CURL_SPACE_SEPARATOR = ' ';
+	private static final String CURL_QUOTE_SEPARATOR = '"';
+	private static final String CURL_COLON_SEPARATOR = ':';
+
 	/**
 	 * Credentials Interface
 	 */
 	public interface ICredentials
```

Comment:
Add something to the readme as this is a breaking change.
I also think we should create a change log.

Reply:
https://github.com/financialforcedev/ffhttp-core/blob/feature/named-credentials/CHANGELOG.md